Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -8,29 +10,46 @@ namespace
 BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
 : initialStateOfChargePercent_(initialStateOfChargePercent)
 {
+
 }
 
 BatteryStateOfChargeService::~BatteryStateOfChargeService()
 {
+
 }
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return BATTERY_AMP_HOUR_CAPACITY - (initialStateOfChargePercent_ * BATTERY_AMP_HOUR_CAPACITY);
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+
+    if(current_> 0)
+        return false;
+    return true;
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+   int charging = (totalAmpHoursUsed()/current_)*3600000;
```

Comment:
`charging` and `depleting` can be 1 variable that is updated in the if statements.